app/controllers/users_controller.rb

```diff
@@ -47,22 +47,30 @@ def create
   end
 
   def update
+    @password_verification = user_verification_params
     @user = current_user
     @user = User.find_by(username: params[:id]) if params[:id] && current_user && current_user.role == "admin"
-    @user.attributes = user_params
-    @user.save({}) do |result|
-      if result
-        if session[:openid_return_to] # for openid login, redirects back to openid auth process
-          return_to = session[:openid_return_to]
-          session[:openid_return_to] = nil
-          redirect_to return_to
+    if @user.valid_password?(user_verification_params["current_password"]) || user_verification_params["ui_update"] != "0"
+      # correct password
+      @user.attributes = user_params
+      @user.save({}) do |result|
+        if result
+          if session[:openid_return_to] # for openid login, redirects back to openid auth process
+            return_to = session[:openid_return_to]
+            session[:openid_return_to] = nil
+            redirect_to return_to
+          else
+            flash[:notice] = I18n.t('users_controller.successful_updated_profile') + "<a href='/dashboard'>" + I18n.t('users_controller.return_dashboard') + " &raquo;</a>"
+            return redirect_to "/profile/" + @user.username + "/edit"
+          end
         else
-          flash[:notice] = I18n.t('users_controller.successful_updated_profile') + "<a href='/dashboard'>" + I18n.t('users_controller.return_dashboard') + " &raquo;</a>"
-          return redirect_to "/profile/" + @user.username + "/edit"
+          render :template => 'users/edit'
         end
-      else
-        render :template => 'users/edit'
       end
+    else
+      # incorrect password
+      flash[:error] = "Current Password entered is not correct!"
```

Comment:
It should be ``Current password is incorrect``.
Please fix this minor error.

Reply:
done